lib/jxl/decode_test.cc

```diff
@@ -1600,32 +1600,20 @@ TEST(DecodeTest, PixelTestOpaqueSrgbLossy) {
     jxl::Span<const uint8_t> span0(pixels.data(), pixels.size());
     jxl::CodecInOut io0;
     io0.SetSize(xsize, ysize);
-    EXPECT_TRUE(ConvertFromExternal(
-        span0, xsize, ysize, color_encoding0,
-        /*has_alpha=*/false, false, 16, format_orig.endianness,
-        /*flipped_y=*/false, /*pool=*/nullptr, &io0.Main(), /*float_in=*/false,
-        /*align=*/0));
+    EXPECT_TRUE(ConvertFromExternal(span0, xsize, ysize, color_encoding0, 3,
+                                    false, 16, format_orig.endianness,
+                                    /*flipped_y=*/false, /*pool=*/nullptr,
+                                    &io0.Main(), /*float_in=*/false,
+                                    /*align=*/0));
 
     jxl::ColorEncoding color_encoding1 = jxl::ColorEncoding::SRGB(false);
     jxl::Span<const uint8_t> span1(pixels2.data(), pixels2.size());
     jxl::CodecInOut io1;
-    if (channels == 4) {
-      io1.metadata.m.SetAlphaBits(8);
```

Comment:
Why is this code gone?

Reply:
I think it dates back to a time when ConvertFromExternal would not work when the buffer is RGBA while the image does not have alpha, so this probably was a workaround to fix that. But ConvertFromExternal just ignores the A if the image does not have alpha, so this workaround is not needed anymore.